skfda/exploratory/stats/covariance/_parametric_gaussian.py

```diff
@@ -48,8 +53,14 @@ def fit(
         regressor.fit(grid_points, data_matrix.T)
 
         # TODO: Skip cov computation?
-        self.covariance_ = X.cov().copy(
-            data_matrix=regressor.kernel_(grid_points)[np.newaxis, ...],
+        # TODO: Use a user-public structure to represent the covariance,
+        #  instead of a Callable object
+        self.covariance_ = X.cov()
+        assert isinstance(self.covariance_, EmpiricalGrid)
+        self.covariance_.cov_fdata = self.covariance_.cov_fdata.copy(
```

Comment:
Copying the `FData` allows to set the `data_matrix` correctly. Directly setting the attribute does not work, since the dimensions must be adjusted. Alternatively, one may do the same as implemented in the constructor of `FData`, which is calling `_int_to_real(np.atleast_2d(data_matrix))`.